app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt

```diff
@@ -276,6 +282,32 @@ class ConversationMessagesViewModel @Inject constructor(
         }
     }
 
+    fun updateImageOnFullscreenMode(message: UIMessage?) {
+        imageMessageOnFullscreen = message
+    }
+
+    fun checkPendingActions(onMessageReply: (UIMessage) -> Unit) = viewModelScope.launch {
+        savedStateHandle.getBackNavArg<Pair<String, String>>(EXTRA_ON_MESSAGE_REACTED)?.let { (messageId, emoji) ->
+            toggleReaction(messageId, emoji)
+        }
+        savedStateHandle.getBackNavArg<String>(EXTRA_ON_MESSAGE_REPLIED)?.let { messageId ->
+            imageMessageOnFullscreen?.let { onFullscreenMessage ->
+                updateImageOnFullscreenMode(null) // We need to reset the imageOnFullscreenMode as we handle it here
+                if (onFullscreenMessage.messageHeader.messageId == messageId) {
+                    onMessageReply(onFullscreenMessage)
+                }
+            }
+        }
+        savedStateHandle.getBackNavArg<Pair<String, Boolean>>(EXTRA_ON_MESSAGE_DETAILS_CLICKED)?.let { (messageId, isSelfAsset) ->
+            imageMessageOnFullscreen?.let { onFullscreenMessage ->
+                updateImageOnFullscreenMode(null) // We need to reset the imageOnFullscreenMode as we handle it here
```

Comment:
why we need to do that ? cannot read it from the code ?

Reply:
check comment above.